### ISSUE TITLE
Question about training the Transient Prediction Network

### ISSUE THREAD

**Issue by harry-graves** [facebookresearch/egolifter#10]
Hi there! Firstly, thanks for all the great work!

My question is regarding the Transient Prediction Network. From what I understand, for each input image, it outputs a probability mask whose values indicate the probability of each pixel belonging to a transient object.

How is this network trained? From what I've read, it is trained during the 3D reconstruction process, as the mask is included in the reconstruction loss function. I can't quite wrap my head around how this works though; without the Transient Prediction Network, the reconstruction loss aims to reconstruct the RGB image using Gaussians, including dynamic objects such as the subject's hands. Surely not including the hands would result in increased reconstruction loss? I understand that the pixels corresponding to the hands are weighted by the mask to have less impact on the reconstruction loss, but how do we arrive at this mask in the first place? I'm stuck in a loop!

Am I missing something? Is the Transient Prediction Network actually pretrained, and simply provides a mask for each RGB image that is then used to weigh the reconstruction loss during training? If so, what data was this trained on?

Any help is appreciated, and again, thanks for all the great work!

**Comment by georgegu1997** [comment]
Thanks for your interest and the question! 

Your understanding is basically correct. The transient prediction network is an image-to-image segmentation network whose backbone is pretrained on ImageNet and that's it. 

After that, it is trained together with the 3DGS reconstruction pipeline in a **self-supervised** fashion. The underlying intuition is roughly that: the transient objects are not view consistent and are hard for 3DGS to reconstruct, and in such case, it will "turn to" the transient prediction network to filter them out from reconstruction by predicting high transient probability on them. For static objects, they are view consistent and easy to reconstruct and thus it does not need to treat them as transient. (Note that it has a regularization on the output in Eq 4).

**Comment by harry-graves** [comment]
Thank you for the reply. It seems I wasn't accounting for **view consistency**. The transient prediction network learns to filter out certain pixels in the jth image so that the reconstruction loss across all images decreases, not just the jth image. I guess the loss function only summing over pixels in the jth image, rather than also summing over all images j = 1 ... M, threw me.

Really appreciate it!
